core/imds/src/main/java/software/amazon/awssdk/imds/Ec2MetadataRetryPolicy.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.imds;
+
+import java.util.Objects;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.retry.RetryMode;
+import software.amazon.awssdk.core.retry.backoff.BackoffStrategy;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Interface for specifying a retry policy to use when evaluating whether or not a request should be retried. The
+ * {@link #builder()}} can be used to construct a retry policy from SDK provided policies or policies that directly implement
+ * {@link BackoffStrategy} .
+ *
+ * When using the {@link #builder()} the SDK will use default values for fields that are not provided. The default number of
+ * retries and condition is based on the current {@link RetryMode}.
```

Comment:
Looks like Javadocs need to be updated accordingly, e.g., there is no RetryMode. Can we also add HTML tag `<p>` between each paragraph.

Reply:
Noted.